server/server.go

```diff
@@ -634,6 +636,19 @@ func (s *Server) v1DiagnosticsGet(w http.ResponseWriter, r *http.Request) {
 	writer.JSON(w, 200, resp, pretty)
 }
 
+func (s *Server) unversionedGetHealth(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	// Create very simple query that binds a single variable.
+	rego := rego.New(rego.Query("x = 1"))
+	// Run evaluation.
+	rs, err := rego.Eval(ctx)
+	if err != nil {
+		writer.ErrorAuto(w, err)
+		return
+	}
+	writer.JSON(w, http.StatusOK, rs[0].Bindings, false)
```

Comment:
As far as the response goes, I'd like to make this consistent with other CNCF projects. What are projects like Kubernetes, Envoy, etc. returning in response to health checks? I could imagine a simple check here that verifies the bindings are correct and just returns 200 or not based on that. What are other projects doing?

Reply:
I'm interested to know how _other_ projects are implementing in their health endpoints. If we're not sure, I'd rather just return HTTP 200 (or not) depending on the result of the check (instead of returning the bindings). The implementation can check that the result set is `[{x: 1}]` and return 5xx if it's not.

Reply:
For what it's worth, there's a [common health endpoint definition](https://github.com/grpc/grpc/blob/master/doc/health-checking.md) in gRPC: even a dysfunctional server response would be (the GRPC equivalent of) HTTP 200, but the response payload gives a hint on what's going on. Also, it declares
> A client can declare the server as unhealthy if the rpc is not finished after some amount of time.

to cover those cases where the server can't even report its own unhealthiness.

For concrete examples of other projects, this is what Dex does: [handlers.go](https://github.com/dexidp/dex/blob/4329f407dcfefad56aa81b4d3da1a8e487bd023e/server/handlers.go#L23-L51), when its attempts to use its storage subsystem fail, it'll spew out a 500.

Reply:
I can certainly return a 500 if the query return is not [{x: 1}]. The scope of the feature request seems to me an orchestrator performing a liveness probe for OPA and *not* OPA checking itself. These are very different things, therefore I gave the k8s example because OPA would most likely be checked for liveness by K8 core and not check its own subsystems.

Reply:
Let's make the check return 200 if there are no errors. I don't have a strong opinion about whether it has to check for x=1 in the binding list, it just seemed like an obvious check to do to ensure that the OPA is healthy.

I just want to remove the bindings from the response body as that represents an extension to the API which we always want to avoid if possible (it's always possible to extend the API later if we want to, but you can't take it away once it's there.)